### PR TITLE
AyaListRepo and Usecase to get ayas

### PR DESCRIPTION
**Pull request by helloseyedjafari**

### CODE REVIEW

File: core/src/commonMain/kotlin/com/islamversity/core/Mapper.kt [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+import kotlinx.coroutines.flow.map
+
 interface Mapper<T, R>{
     fun map(item : T) : R
 }
 
 fun <T, R> Mapper<T, R>.listMap(items: List<T>) : List<R> =
     items.map { this.map(it) }
+
+fun <T, R> Flow<T>.mapWith(mapper : Mapper<T, R>) : Flow<R> =
```
Comment (rezaiyan) [inline core/src/commonMain/kotlin/com/islamversity/core/Mapper.kt:13]: cool 👍

File: daggercore/src/main/java/com/islamversity/daggercore/modules/domain/AyaModule.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        mapper: Mapper<Aya, AyaRepoModel>,
+    ): AyaListRepo =
+        AyaListRepoImpl(
+            dataSource,
+            mapper
+        )
+
+    @JvmStatic
+    @Provides
+    fun provideAyaEntityRepoMapper(): Mapper<Aya, AyaRepoModel> = AyaEntityRepoMapper()
```
Comment (rezaiyan) [inline daggercore/src/main/java/com/islamversity/daggercore/modules/domain/AyaModule.kt]: Would you like to separate mapper providers into another module

**Review by rezaiyan — COMMENTED** [review]
Doing great 👍

File: daggercore/src/main/java/com/islamversity/daggercore/modules/domain/AyaModule.kt (hunk 2 above)
Comment (helloseyedjafari) [reply daggercore/src/main/java/com/islamversity/daggercore/modules/domain/AyaModule.kt]: yeah let's do it!

**Pull request merged**